File: Frontend/src/types/analyticAccount.ts
export type AnalyticType = 'INCOME' | 'EXPENSE'

export interface AnalyticAccount {
  id: number
  name: string
  analytic_type: AnalyticType
  is_active: boolean
  archived_at: string | null
  created_at: string
  updated_at: string
}

export interface AnalyticAccountInput {
  name: string
  analytic_type: AnalyticType
}

File: Backend/app/schemas/analytic_account.py
"""Analytic Account schemas — design doc §3 item 6."""
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import AnalyticType


class AnalyticAccountCreate(BaseModel):
    name: str = Field(min_length=1, max_length=128)
    analytic_type: AnalyticType


class AnalyticAccountUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=128)
    analytic_type: AnalyticType | None = None


class AnalyticAccountOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    analytic_type: AnalyticType
    is_active: bool
    archived_at: datetime | None
    created_at: datetime
    updated_at: datetime
